tests/e2e-k8s/lib/common.sh

```
@@ -0,0 +1,140 @@
+  #    SSH logout; without linger, systemd --user (and thus the rootless runtime)
+  #    can vanish mid-test.
+  if ! loginctl show-user "$(id -un)" 2>/dev/null | grep -q '^Linger=yes'; then
+    warn "user linger is OFF — run: sudo loginctl enable-linger \$USER"
+    warn "  (rootless podman containers will die when your login session ends)"
+    problems=$((problems+1))
+  fi
+
+  # 2. cgroup v2 delegation. Rootless podman with cgroupManager=systemd needs the
+  #    user delegated @set cpuset/cpu/io/memory/pids. This is the single most
+  #    common kind-in-podman breakage. We can only READ this from the user slice.
+  local ucgroup="/sys/fs/cgroup/user.slice/$(id -u).user.slice"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)